contributors/design-proposals/kubectl-unset.md

```diff
@@ -0,0 +1,70 @@
+# Kubectl Unset Subcommand
+
+## Goals
+
+`kubectl unset` is used to remove specific features from objects. To be contrasted with `kubectl set` subcommand, `unset` commands should has the same ability as `set` commands.
+
+## Design
```

Comment:
spend some time in this section explaining why `unset` is different using some concrete examples.  Generally speaking, `set` commands will have more required fields than `unset` commands, since `unset` only needs to identify a thing to be removed, as opposed to defining what the the "thing" is.  This has an impact on auto-complete, help, and validation.

In addition, `kubectl unset` is not mutually exclusive to a `kubectl set --remove` or some such since the `--remove` could actually build the `kubectl unset` options.